src/supla/network/html/button_update.cpp

```diff
@@ -0,0 +1,26 @@
+#include "button_update.h"
+
+#include <supla/network/web_sender.h>
+
+#ifdef ARDUINO_ARCH_ESP32
+#include <HTTPUpdateServer.h>
+HTTPUpdateServer httpUpdater;
+#else
+#include <ESP8266HTTPUpdateServer.h>
+ESP8266HTTPUpdateServer httpUpdater;
+#endif
```

Comment:
This will always create httpUpdater object on ESP32/ESP8266 compilation, even if "button update" is not used in program at all.

Please move this code i.e. to ButtonUpdate contructor and create this object with "new" operator.